lib/matplotlib/cm.py

```diff
@@ -91,13 +92,66 @@ def _warn_deprecated(self):
         )
 
 
+class ColormapRegistry(Mapping):
+    r"""
+    Container for colormaps that are known to Matplotlib by name.
+
+    Read access uses a dict-like interface mapping names to `.Colormap`\s.
+    Returned `.Colormap`.s are copies, so that their modification does not
+    change the global definition of the colormap.
+
+    Additional colormaps can be added via `.ColormapRegistry.register`.
+    """
+    def __init__(self, cmaps):
+        self._cmaps = cmaps
+
+    def __getitem__(self, item):
+        try:
+            return copy.deepcopy(self._cmaps[item])
+        except KeyError:
+            raise KeyError(f"{item!r} is not a known colormap name")
+
+    def __iter__(self):
+        return iter(self._cmaps)
+
+    def __len__(self):
+        return len(self._cmaps)
+
+    def register(self, cmap, *, name=None):
+        """
+        Register a new colormap.
+
+        The colormap name can then be used as a string argument to any ``cmap``
+        parameter in Matplotlib. It is also available in `.pyplot.get_cmap`.
+
+        The colormap registry stores a copy of the given colormap, so that
+        future changes to the original colormap instance do not affect the
+        registered colormap. Think of this as the registry taking a snapshot
+        of the colormap at registration.
+
+        Parameters
+        ----------
+        cmap : matplotlib.colors.Colormap
+           The colormap to register.
+
+        name : str, optional
+           The name for the colormap. If not given, the :attr:`~.Colormap.name`
+           attribute of *cmap* is used.
+        """
+        register_cmap(name, copy.deepcopy(cmap))
```

Comment:
There is a `copy()` method on Colormaps now, so you should be able to call `cmap.copy()` instead and avoid the copy import above.